### PR TITLE
Add Version Number to URL

### PR DESCRIPTION
**Pull request by emaildano**
Explicitly label version of jQuery for security and plugin compatibility.

### CODE REVIEW

**Comment by retlehs** [comment]
> for security and plugin compatibility

please explain this — this PR isn't necessary

**Comment by emaildano** [comment]
A few plugins (iThemes Security Pro for example) cannot verify a safe version of jQuery with sites using Soil.

So, that PR takes this..

`ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js`

and makes it this..

`ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js?ver=1.11.0`

Removing that flag and verifying the version. Otherwise, there's no way of knowing.

**Comment by retlehs** [comment]
seems like a bug with ithemes security pro. they should be able to verify that jquery from the google CDN is a "safe" version of jquery.

**Comment by emaildano** [comment]
That pattern matches what WordPress provides, the one which is removed.

If the purpose of removing it and adding the Google CDN is for speed then your link is incomplete.

Bug, preference, or plugin aside.

**Comment by retlehs** [comment]
a lot of the code in soil **changes wordpress core markup** on purpose

easy fix: don't use the google cdn for jquery

better fix: get ithemes security pro to fix their plugin. it should be intelligent enough to know that the google CDN is safe.

**Comment by emaildano** [comment]
Soil should be intelligent enough to declare it.

How else does a plugin or browser verify a version? It can't read the comments, it can only read the URL and variables it's being served.

Right now it's servicing a blind link.

**Comment by retlehs** [comment]
the URL has the version number in it already...

i already mentioned two options you have.

**Comment by QWp6t** [comment]
> How else does a plugin or browser verify a version?

**Plugin**: Check the URL or retrieve file, parse comment, and cache result.
`/*! jQuery v1.11.3 | (c) 2005, 2015 jQuery Foundation, Inc. | jquery.org/license */`

**Browser**: `$.fn.jquery`
`"1.11.3"`

You can also hook into `wp_enqueue_scripts` with lower priority than 100 and store jquery version number (just as we do), then append it back by setting higher priority on `script_loader_src` hook. That will allow you to keep using this module with a security plugin that's not actually doing anything for security by checking some arbitrary query string value.
